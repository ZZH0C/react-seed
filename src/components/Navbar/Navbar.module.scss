.reset_button:focus{
border-top: 5px #646d81 !important;
padding-top: 10px !important;
}