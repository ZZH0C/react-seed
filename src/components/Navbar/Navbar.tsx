import React from 'react';

type NavbarProps = React.HTMLAttributes<HTMLElement>;

export const Navbar: React.FC<NavbarProps> = ({ children }) => {
  return (
    <nav className="navbar navbar-default navbar-top">
      <div className="container">
        <div className="navbar-header">
          <button
            type="button"
            className="navbar-toggle collapsed"
            data-toggle="collapse"
            data-target=".bs-example-navbar-collapse-1"
          >
            <span className="sr-only">Toggle navigation</span>
            <span className="icon-bar"></span>
            <span className="icon-bar"></span>
            <span className="icon-bar"></span>
          </button>
        </div>
        <div className="collapse navbar-collapse bs-example-navbar-collapse-1">
          <ul className="nav navbar-nav">{children}</ul>
        </div>
      </div>
    </nav>
  );
};
