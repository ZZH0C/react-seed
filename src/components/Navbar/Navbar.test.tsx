import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { Navbar } from './Navbar';

describe('components/ButtonLink', () => {
  const mockProps = {
    children: 'SOME_TEXT',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(shallow(<Navbar>{mockProps.children}</Navbar>)).toMatchSnapshot();
  });
});
