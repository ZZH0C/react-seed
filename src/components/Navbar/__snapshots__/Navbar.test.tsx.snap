// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<nav
  className="navbar navbar-default navbar-top"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <button
        className="navbar-toggle collapsed"
        data-target=".bs-example-navbar-collapse-1"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      className="collapse navbar-collapse bs-example-navbar-collapse-1"
    >
      <ul
        className="nav navbar-nav"
      >
        SOME_TEXT
      </ul>
    </div>
  </div>
</nav>
`;
